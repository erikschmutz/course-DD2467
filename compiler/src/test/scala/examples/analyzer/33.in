type color = YELLOW | RGB of int * int * int;;
let red = RGB(255, 0, 0);;

(*====================
type color : color (YELLOW | RGB of int * int * int)
val red : RGB of int * int * int
- :RGB of int * int * int
====================*)