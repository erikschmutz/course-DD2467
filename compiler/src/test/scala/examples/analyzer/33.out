AnalyzerResult(Constraint(RGB,Some(Tuple(List(Integer(),Integer(),Integer()),List())),List()),Enviroment(List(TypeEntry(Identifier(color),Constraints(color,List(Constraint(YELLOW,None,List()),Constraint(RGB,Some(Tuple(List(Integer(),Integer(),Integer()),List())),List())),List())),ValEntry(Identifier(red),Constraint(RGB,Some(Tuple(List(Integer(),Integer(),Integer()),List())),List())))))